src/syft/core/io/route.py

```diff
@@ -3,24 +3,54 @@
 from typing import final
 
 
+class BaseRoute(object):
+    connection_details = {}
+
+    def configure_connection(self, protocol: str, host: str, port: int):
+        """
+        the route should have connection details embedded in it.
+        so that nodes operators can utilize it to route messages.
+        """
+        self.connection_details.update({
+            'host': host,
+            'protocol': protocol,
+            'port': port
+        })
+
+    def register_broadcast_channel(self, channel_name: str):
```

Comment:
I wonder if from a Route perspective it might be better to create a separate abstraction called a "Group". That way the worker could use channels to talk to the group (according to some naming convention) but if pub/sub wasn't available it could fallback to individually messaging each member of the group.